### DISCUSSION TITLE
Fuzzy searching on a set of fields

### DISCUSSION THREAD

**Discussion by maartenvn** [directus/directus#15314] [Feature Requests]
Right now Directus allows searching using filters or using the search parameter.
This search is not ideal in a lot of cases:
* It does not allow fuzzy searching (you have to enter the exact search term)
* It does not allow filtering multiple fields at once (you can only use `search` for that, which is only on non-related fields)

To fix this I propose a new fuzzy searching system with a few parameters. For example
* `?search=ABBA Dancing Queen`: search term
* `?searchFields=["song.title", "song.artist"]`: specify which fields must be searched
* `?searchType=fuzzy` or `?searchType=exact`: fuzzy or exact searching of the fields
* `?searchOptions={ ... }`: options for the fuzzy searching

A library like [Fuse.JS](https://fusejs.io/api/options.html) can be used to implement this behavior.

**Comment by azrikahar** [comment]
This would be a great enhancement to the current search indeed! I believe this can also tie into the idea of "search adapters" that have been floating around in previous related discussions, such as https://github.com/directus/directus/discussions/12264#discussioncomment-2407051 and https://github.com/directus/directus/discussions/4492#discussioncomment-1760168.

**Comment by rijkvanzanten** [comment]
> A library like [Fuse.JS](https://fusejs.io/api/options.html) can be used to implement this behavior.

I wish that was the case.. The search has to happen in SQL, otherwise we would have to copy the whole table into JS memory first, and do the search there (with something like Fuse). On tables with hundreds of thousands of records, this will easily cause major issues.

There isn't really a standardized way to do a fusejs style fuzzy search in SQL though, so I think this is indeed where search adapters come in, so you can hook up a third party searching platform (like algolia) to handle the more advanced searching capabilities

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
